word_game.rb

```diff
@@ -0,0 +1,328 @@
+  def has_letter?(user_input)
+    return @secret_word.downcase.include?(user_input)
+  end
+
+end
+
+
+######################################################################
+# FUNCTIONALITY METHODS
+
+
+# Method to get user input regarding the desired difficulty of the game to be
+# played and returns it, after validating it as a valid difficulty option.
+def get_difficulty_level
+  difficulty = ""
+  until difficulty == "easy" ||  difficulty == "medium" || difficulty == "hard"
+    print "Please select a difficulty level (easy, medium, hard) > ".blue
+    difficulty = gets.chomp.strip
+  end
+  return difficulty
+end
+
+# Method to print text in blue.
+# All the system/program related text should be in blue if prompting the user for
+# an input or giving instructions.
+def print_in_blue(text_to_print)
+  print "#{text_to_print}".blue
+end
+
+# Method to handle the user input if equals to 'exit'
+# It will prompt the user if really want to quit the program
+# and if answer is 'y' then it will exit.
+def handle_input(input_letter)
+  if input_letter == "exit"
+    print "Are you sure you want to exit? (y/n) > ".blue
+    exit if gets.chomp.downcase == "y"
+  end
+end
+
+# Method to print a welcome message and game directions, explainning the difficult level
+# atrribute and how to use it.
+def print_directions
+  puts
+  print_in_blue("Welcome to the Word Guess Game!\n\n")
+  print_in_blue("When prompted for a letter, type #{"exit".bold} to quit the program or")
+  print_in_blue("#{"restart".bold} to give up and start a new game.")
+  print_in_blue("This game has three difficulty levels with different themes:")
+  print_in_blue("#{'Easy'.bold} is random movie quotes.")
+  print_in_blue("#{'Medium'.bold} is random dishes.")
+  print_in_blue("#{'Hard'.bold} is random color.\n\n")
+end
+
+# Method to print string in blue and centralized on the user terminal's screen.
+def print_in_blue(text_to_print)
+  puts "#{text_to_print}".center(60).blue
+end
+
+
+# Method to initialize the game by printing directions, prompting user for a difficulty
+# level and then initiating the class Game. It willl run until the game is over
+# and handle game options like restart it self or quit the application.
+def play_game
+  print_directions
+  difficulty_level = get_difficulty_level
+  word_game = Game.new(Word.new(difficulty_level))
+  until word_game.game_over?
+    word_game.print_status
+    input_letter = word_game.get_input
+    handle_input(input_letter)
+    if input_letter == "restart"
+      word_game.give_up
+      break
+    end
+    next if !word_game.valid_input(input_letter)
+    word_game.process_guess_input(input_letter)
+  end
+  word_game.display_end_of_game
+end
+
+# Method to prompt the user for an yes or no answer on the matter of starting the game all
+# over again after the game is over (when the user wins or looses).
+def play_again?
+  print "\nThe game is over! Would you like to play again? (Y/N) >".center(60).blue
```

Comment:
You should, "trap," the user in a loop that doesn't exit until they enter "Y" or "N".